core/src/main/java/org/mule/runtime/core/api/config/FeatureFlaggingRegistry.java

```diff
@@ -67,9 +67,9 @@ private FeatureFlaggingRegistry() {}
    * @param condition This predicate will be evaluated at deployment time. The {@link MuleContext} corresponds to the context that
    *        is being created for this application.
    */
-  public void registerFeature(String feature, Predicate<MuleContext> condition) {
-    if (isNullOrEmpty(feature)) {
-      throw new MuleRuntimeException(createStaticMessage("Invalid feature name"));
+  public void registerFeature(Feature feature, Predicate<MuleContext> condition) {
+    if (feature == null) {
+      throw new MuleRuntimeException(createStaticMessage("Feature can not be null"));
```

Comment:
extract test to constant.
shouldn't this be an `IllegalArgumentException` instead of a `MuleRuntimeException`?